src/pages/students/forms/paymentDataForm/PaymentData.tsx

```diff
@@ -1,77 +1,69 @@
         InputsContainer,
-        Text} from "../styles/PaymentData.styles";
-import { PaymentPlanName } from "../../../app/types/models/PaymentPlanName";
+        Text} from "./PaymentData.styles";
+import { PaymentPlanName } from "../../../../app/types/models/PaymentPlanName";
 import { useForm } from "react-hook-form";
 import { yupResolver } from "@hookform/resolvers/yup";
 import * as yup from "yup";
-import { paymentDataScheme } from "../schemes/PaymentData.scheme";
-import { ErrorMessage } from "../../../components/header/ErrorMessage";
-import { useNavigate } from "react-router";
-import { setStudentData } from "../StudentRegistrationFormSlice";
-import type { RootState } from "../../../app/store/store";
-import { useDispatch, useSelector } from "react-redux";
+import { paymentDataScheme } from "./PaymentData.scheme";
+import { ErrorMessage} from "../../../../components/error_message/ErrorMessage"
+import type { PaymentDataProps } from "../../create-student/CreateStudent";
 
+interface FormProp {
+  createStudentCall: (data: PaymentDataProps) => void;
+  paymentData:PaymentDataProps | undefined;
+  stepBack:()=>void;
+}
 
-function PaymentData() {
+function PaymentData({
+  createStudentCall,
+  paymentData,
+  stepBack
+}: FormProp) {
 
-  type FormData = yup.InferType<typeof paymentDataScheme>;
 
-  const dispatch = useDispatch();
+  type FormData = yup.InferType<typeof paymentDataScheme>;
     
-  const studentRegistrationForm = useSelector((state: RootState) => state.studentRegistrationForm);
+  const studentRegistrationForm = paymentData ||
+  {
+    paymentPlanName:"",
+    classPrice:null,
+    extraClasses:null,
+    paymentDay:undefined,
+  };
   
   const PlanTypeNameArray = Object.values(PaymentPlanName);
 
   const {
     register,
     handleSubmit,
     watch,
-    getValues, 
     formState: { errors },
   } = useForm<FormData>({
     resolver: yupResolver(paymentDataScheme) as any,
-    defaultValues:studentRegistrationForm,
+    defaultValues:{...studentRegistrationForm},
   });
 
   const PaymentPlanNameSelected = watch("paymentPlanName");
 
-  const navigate = useNavigate();
-
-  const normalizePaymentData = (data: FormData) => ({
-    ...data,
-    extraClasses: data.extraClasses ?? undefined,
-    classPrice: data.classPrice ?? undefined,
-    paymentDay: data.paymentDay ?? undefined,
-  });
-
   const onSubmit = (paymentData: FormData) => {
-    const normalizedData = normalizePaymentData(paymentData);
-    dispatch(setStudentData(normalizedData)); 
-    navigate("/datos-del-plan");
-  };
-
-  const stepBack = () => {
-    const paymentData = getValues();
-    const normalizedData = normalizePaymentData(paymentData);
-    dispatch(setStudentData(normalizedData));
-    navigate("/datos-del-alumno");
+     createStudentCall({
+      paymentPlanName: paymentData.paymentPlanName,
+      paymentDay: paymentData.paymentDay,
+      classPrice:paymentData.classPrice,
+      extraClasses:paymentData.extraClasses
+    })
```

Comment:
Aca podrias hacer lo siguiente creo


```suggestion
     createStudentCall({ ...paymentData })
```